sources/api/migration/migrations/v1.14.1/aws-partition-setting/src/main.rs

```diff
@@ -0,0 +1,24 @@
+use migration_helpers::common_migrations::{AddPrefixesMigration, AddSettingsMigration};
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added a new setting for AWS partition.
+fn run() -> Result<()> {
+    if cfg!(variant_platform = "aws") {
+        migrate(AddSettingsMigration(&["settings.aws.partition"]))
+    } else {
+        // Non-AWS variants did not have any AWS setting until this point,
```

Comment:
That's not actually true anymore. Since v.1.11.0 we've needed `settings.aws` to support AWS IAM Roles Anywhere.

https://github.com/bottlerocket-os/bottlerocket/pull/2542/commits/cedda93d79c849948c063dccedc076d1a05580fe

Reply:
Ah, in that case I'll replace the else with a `NoOpMigration`.